.github/workflows/build.yaml

```diff
@@ -240,15 +240,15 @@ jobs:
           git submodule foreach --recursive 'git rev-parse HEAD || rm -rf $PWD'
           $LASTEXITCODE = 0
 
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           fetch-depth: '0'
           submodules: 'recursive'
 
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@v5
         with:
-          # force a 3.7 is ok, build.py will handle actual python env
-          python-version: 3.7
+          # force a 3.12 is ok, build.py will handle actual python env
+          python-version: 3.12
```

Comment:
In general, my experience of python 3.x is that:
- later versions can run code that runs on earlier versions (backwards compatible)
- earlier versions often miss features of later versions, and cannot run code designed to run on later versions

Therefore, on the whole, if we are going to test on a single python version, we should test on the minimum supported version, I feel.

(Again, I'm just a random bystander, and have no abiltiy to merge, just to be clear).

Reply:
Hello, it is because python 3.12 is used as baseline nowadays